src/api/venue.ts

```diff
@@ -110,12 +112,32 @@ export const fetchDirectChildVenues = async (
       async (venueIdsChunk: string[]) => {
         const childVenuesSnapshot = await getVenueCollectionRef()
           .where("parentId", "in", venueIdsChunk)
+
           .withConverter(anyVenueWithIdConverter)
           .get();
 
-        return childVenuesSnapshot.docs
-          .filter((docSnapshot) => docSnapshot.exists)
-          .map((docSnapshot) => docSnapshot.data());
+        const venueWithEvents = Promise.all(
+          childVenuesSnapshot.docs
+            .filter((docSnapshot) => docSnapshot.exists)
+            .map((docSnapshot) => docSnapshot.data())
+            .map(async (venue) => {
+              const second = await getVenueCollectionEventsRef(venue.id)
+                .withConverter(anyVenueEventWithIdConverter)
+                .get();
+
+              const rest = second.docs
+                .filter((docSnapshot) => docSnapshot.exists)
+                .map((docSnapshot) => docSnapshot.data());
+              return {
+                ...venue,
+                ...(rest.length
+                  ? { events: rest as WithId<VenueEvent>[] }
+                  : null),
+              };
+            })
+        );
+
+        return venueWithEvents;
```

Comment:
looks complicated... are you extending the venue with its events here? 
not sure we should do that. Can we handle that separately?